Plots/Station/NCL_station_3.py

```diff
@@ -0,0 +1,157 @@
+"""
+NCL_station_3.py
+================
+This script illustrates the following concepts:
+   - Drawing station numbers on a map, and removing ones that overlap
+   - Attaching lots of text strings to a map
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/station_3.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/station_3_1_lg.png and https://www.ncl.ucar.edu/Applications/Images/station_3_2_lg.png
+"""
+
+###################################################
+# Import packages:
+
+import numpy as np
+import pandas as pd
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+from matplotlib import pyplot as plt
+import matplotlib.ticker as mticker
+
+import geocat.datafiles as gdf
+
+###################################################
+# Generate data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds = pd.read_csv(
+    gdf.get('ascii_files/istasyontablosu_son.txt'),
+    delimiter='\\s+',
+    names=['index', 'station', 'year1', 'year2', 'number', 'lat', 'lon'])
+
+npts = len(ds)  # get number of points
+
+# Extract variables
+no = ds.index + 1  # +1 because Pandas' RangeIndex defaults start with 0
+lat = ds.lat
+lon = ds.lon
+
+##############################################################################
+# Helper function to add plot elements to the axes
+
+
+def create_axes(maintitle):
+
+    # Generate figure (set its size (width, height) in inches)
+    fig = plt.figure(figsize=(12, 6.5))
+
+    # Generate axes
+    ax = plt.axes(projection=ccrs.Mercator())
+
+    # Set extent to show particular area of the map
+    ax.set_extent([25.5, 45.2, 35.5, 42.5], ccrs.PlateCarree())
+
+    # Add state boundaries other lake features
+    ax.add_feature(cfeature.LAND, facecolor='none', edgecolor='gray')
+
+    # Draw gridlines
+    gl = ax.gridlines(crs=ccrs.PlateCarree(),
+                      draw_labels=True,
+                      dms=False,
+                      x_inline=False,
+                      y_inline=False,
+                      linewidth=1,
+                      color="black",
+                      alpha=0.25)
+
+    # Set frequency of gridlines in the x and y directions
+    gl.xlocator = mticker.FixedLocator(np.arange(26, 45, 2))
+    gl.ylocator = mticker.FixedLocator(np.arange(36, 43, 1))
```

Comment:
Similarly, the original NCL plot has tick marks while this recreation does not. It seems like you're creating tick marks here and removing them unnecessarily later on. 

Also, a lot of this helper functions looks like it could be accomplished with -viz functions instead of like this.

Reply:
@anissa111 I wasn't able to use viz function set_axes_limits_and_ticks because the projection used for this example is Mercator, so the data is in UTM coordinates, but the latitude and longitude datasets are in degrees. So when I tried to directly set ticks to x axis and y axis, the locations of ticks are wrong because they do not correspond to the Mercator projection's coordinate system. I checked with the other examples that use another projection, [proj_2](https://geocat-examples.readthedocs.io/en/latest/gallery/MapProjections/NCL_proj_2.html#sphx-glr-gallery-mapprojections-ncl-proj-2-py), [native_2](https://geocat-examples.readthedocs.io/en/latest/gallery/MapProjections/NCL_native_2.html#sphx-glr-gallery-mapprojections-ncl-native-2-py), they both use gridlines as a workaround. [native_1](https://geocat-examples.readthedocs.io/en/latest/gallery/MapProjections/NCL_native_1.html#sphx-glr-gallery-mapprojections-ncl-native-1-py) is another example that uses another projection and has to add tickmarks and labels with gridlines. I removed the gridlines and keep the labels to make it similar to the original plot, but both native_1 and native_2 keep the gridlines even though the original plots don't. Let me know if this makes sense to you and if there is a better way to do this!

Reply:
Oh, okay! I'll look into a better way. You should add this workaround to the "concepts illustrated" section!

Reply:
I found the function transform_points in Cartopy that can transform points between coordinate systems.